eval/src/tests/instruction/generic_join/generic_join_test.cpp

```diff
@@ -113,15 +105,19 @@ TEST(GenericJoinTest, dense_join_plan_can_be_executed) {
 TEST(GenericJoinTest, generic_join_works_for_simple_and_fast_values) {
     ASSERT_TRUE((join_layouts.size() % 2) == 0);
     for (size_t i = 0; i < join_layouts.size(); i += 2) {
-        TensorSpec lhs = spec(join_layouts[i], Div16(N()));
-        TensorSpec rhs = spec(join_layouts[i + 1], Div16(N()));
-        for (auto fun: {operation::Add::f, operation::Sub::f, operation::Mul::f, operation::Div::f}) {
-            SCOPED_TRACE(fmt("\n===\nLHS: %s\nRHS: %s\n===\n", lhs.to_string().c_str(), rhs.to_string().c_str()));
-            auto expect = ReferenceOperations::join(lhs, rhs, fun);
-            auto simple = perform_generic_join(lhs, rhs, fun, SimpleValueBuilderFactory::get());
-            auto fast = perform_generic_join(lhs, rhs, fun, FastValueBuilderFactory::get());
-            EXPECT_EQ(simple, expect);
-            EXPECT_EQ(fast, expect);
+        auto l = join_layouts[i];
+        auto r = join_layouts[i+1];
+        for (TensorSpec lhs : { l.gen(), l.cells_double().gen() }) {
+            for (TensorSpec rhs : { r.gen(), r.cells_double().gen() }) {
```

Comment:
the call to r.gen() will produce double cells the second time, which is probably not the intent.